### ISSUE TITLE
Problem with generating Sheets list

### ISSUE THREAD

**Issue by IanMahaffey60** [Jannik-Schilling/generate_swmm_inp#38]
My QGIS was having an issue grabbing lists of the sheets in the function "read_data_from_table_direct". 

I applied information from the following forums:

https://stackoverflow.com/questions/17977540/pandas-looking-up-the-list-of-sheets-in-an-excel-file
https://stackoverflow.com/questions/64264563/attributeerror-elementtree-object-has-no-attribute-getiterator-when-trying

I added a line that said:

`sheets = pd.read_excel(file,None,engine='openpyxl').keys()`

and now the function looks like:

```
def read_data_from_table_direct(file, sheet=0):
    '''reads curves or other tables from excel or csv'''
    filename, file_extension = os.path.splitext(file)
    sheets = pd.read_excel(file,None,engine='openpyxl').keys()
    if file_extension == '.xlsx' or file_extension == '.xls' or file_extension == '.ods':
        if sheet == 0:
            data_df = pd.read_excel(file,sheet_name = sheet, engine='openpyxl')
        else:
            #print ('sheet not 0')
            if sheet in sheets:
                data_df = pd.read_excel(file,sheet_name = sheet, engine='openpyxl')
            elif str(sheet).upper() in sheets:
                data_df = pd.read_excel(file,sheet_name = str(sheet).upper(), engine='openpyxl')
            elif str(sheet).lower() in sheets:
                data_df = pd.read_excel(file,sheet_name = str(sheet).lower(), engine='openpyxl')
            elif str(sheet).capitalize() in sheets:
                data_df = pd.read_excel(file,sheet_name = str(sheet).capitalize(), engine='openpyxl')
            else:
                data_df = pd.DataFrame()
    if file_extension == '.csv':
        data_df = pd.read_csv(file)
    return data_df
```

**Comment by Jannik-Schilling** [comment]
hey @IanMahaffey60 ,
Thank you for reporting this issue. Which version of QGIS, python and pandas are you using? I'll try your suggestion in the next plugin version to fix this issue

**Comment by IanMahaffey60** [comment]
QGIS Version 3.26.0
Python Version 3.9
Pandas Version 1.1.3
(All the standard packages from the latest QGIS update)

Thank you!

**Comment by Jannik-Schilling** [comment]
possibly fixed with [bad5088](https://github.com/Jannik-Schilling/generate_swmm_inp/commit/bad5088eb878ebc11589574d44db1622ef1f99ee).

maybe @IanMahaffey60  you could try the new version https://github.com/Jannik-Schilling/generate_swmm_inp/tree/dev_v_0_23

**Comment by Jannik-Schilling** [comment]
70420f6e6ff2c749c98c1410aab8787b342a1132
